### PR TITLE
Remove Server Header from buildbot master

### PR DESCRIPTION
**Pull request by oxr463**
Removes `Server: TwistedWeb/19.2.1` from all responses.

Signed-off-by: Lucas Ramage <ramage.lucas@protonmail.com>

Closes: https://github.com/buildbot/buildbot/issues/5081

See also: https://stackoverflow.com/q/20304287/8507637

## Contributor Checklist:

* [x] I have updated the unit tests
* [x] I have created a file in the `master/buildbot/newsfragments` directory (and read the `README.txt` in that directory)
* [x] <s>I have updated the appropriate documentation</s> N/A

### CODE REVIEW

**Comment by oxr463** [comment]
@tardyp please review.

**Comment by tardyp** [comment]
looks good. Please add unit test and news fragment.

**Comment by oxr463** [comment]
https://github.com/oxr463/buildbot/blob/remove-server-header/master/buildbot/test/unit/test_www_service.py#L212

**Comment by tardyp** [comment]
> https://github.com/oxr463/buildbot/blob/remove-server-header/master/buildbot/test/unit/test_www_service.py#L212

Is this a question? there is still no unit test to test the absence of Server header.

**Comment by oxr463** [comment]
> 
> 
> > https://github.com/oxr463/buildbot/blob/remove-server-header/master/buildbot/test/unit/test_www_service.py#L212
> 
> Is this a question? there is still no unit test to test the absence of Server header.

Nope, just a personal note for myself. Still trying to find where things are. :)

**Comment by oxr463** [comment]
@tardyp please review.

File: master/buildbot/test/unit/test_www_service.py [hunk 1]
```diff
@@ -274,3 +274,20 @@ def getHost(self):
                              algorithms=[service.SESSION_SECRET_ALGORITHM])
         self.assertEqual(decoded['user_info'], {'anonymous': True})
         self.assertIn('exp', decoded)
+
+    def test_absentServerHeader(self):
+
+        class FakeChannel:
```
Comment (tardyp) [inline master/buildbot/test/unit/test_www_service.py]: please factorize the FakeChannel

Comment (oxr463) [reply master/buildbot/test/unit/test_www_service.py]: Pardon? What is meant by that?

Comment (tardyp) [reply master/buildbot/test/unit/test_www_service.py]: The FakeChannel is copy paste from the previous test, so you could just put that class in the top level of that file, and share it between the two tests.

**Comment by tardyp** [comment]
There are a few issues reported by the CI, can you fix them?

**Comment by oxr463** [comment]
> 
> 
> There are a few issues reported by the CI, can you fix them?

This is done.

**Comment by oxr463** [comment]
> 
> 
> There are a few issues reported by the CI, can you fix them?

This is done.

**Comment by oxr463** [comment]
> 
> 
> There are a few issues reported by the CI, can you fix them?

This is done.

**Comment by p12tic** [comment]
@oxr463 Don't bother changing the PR to rerun tests, they're really unstable lately. I'll take care of the PR.

**Comment by tardyp** [comment]
thanks!

**Pull request merged**